pkg/tfshim/sdk-v2/upgrade_state.go

```diff
@@ -49,8 +49,8 @@ func upgradeResourceStateGRPC(
 		RawState: &tfprotov5.RawState{JSON: jsonBytes},
 		Version:  version,
 	})
-	if err != nil {
-		return cty.Value{}, nil, err
+	if uerr := handleDiagnostics(ctx, resp.Diagnostics, err); uerr != nil {
```

Comment:
This is the essential change. Do not ignore diagnostics, including errors, fail fast here.